packages/fxa-shared/subscriptions/metadata.ts

```diff
@@ -169,3 +172,152 @@ export const getProductSupportApps =
+  useFirestoreProductConfigs: boolean
+) => {
+  if (useFirestoreProductConfigs) {
+    const urls = flattenProductConfig(plan, userLocales).urls;
+    return urls as ProductConfigUrlConfig;
+  } else {
+    const planMetadataConfig = productDetailsFromPlan(plan, userLocales);
+
+    return {
+      termsOfService:
+        planMetadataConfig.termsOfServiceURL ||
+        DEFAULT_PRODUCT_DETAILS.termsOfServiceURL!,
+      termsOfServiceDownload:
+        planMetadataConfig.termsOfServiceDownloadURL ||
+        DEFAULT_PRODUCT_DETAILS.termsOfServiceDownloadURL!,
+      privacyNotice:
+        planMetadataConfig.privacyNoticeURL ||
+        DEFAULT_PRODUCT_DETAILS.privacyNoticeURL!,
+      privacyNoticeDownload:
+        planMetadataConfig.privacyNoticeDownloadURL ||
+        DEFAULT_PRODUCT_DETAILS.privacyNoticeDownloadURL!,
+      cancellationSurvey: planMetadataConfig.cancellationSurveyURL,
+      download:
+        plan.product_metadata?.downloadURL || plan.plan_metadata?.downloadURL,
+    };
+  }
+};
+
+export const webIconConfigFromProductConfig = (
+  plan: AbbrevPlan,
+  userLocales: readonly string[] = [DEFAULT_LOCALE],
+  useFirestoreProductConfigs: boolean
+) => {
+  if (useFirestoreProductConfigs) {
+    const planConfiguration = getPlanProductConfig(plan);
+    const urls = flattenProductConfig(plan, userLocales)
+      .urls! as ProductConfigUrlConfig;
+
+    return {
+      webIcon: urls.webIcon,
+      webIconBackground: planConfiguration.styles?.webIconBackground,
+    };
+  } else {
+    const combinedMetadata = metadataFromPlan(plan as Plan);
+
+    return {
+      webIcon: combinedMetadata.webIconURL,
+      webIconBackground: combinedMetadata.webIconBackground,
+    };
+  }
+};
+
+export const uiContentFromProductConfig = (
+  plan: AbbrevPlan,
+  userLocales: readonly string[] = [DEFAULT_LOCALE],
+  useFirestoreProductConfigs: boolean
+) => {
+  if (useFirestoreProductConfigs) {
+    return flattenProductConfig(plan, userLocales).uiContent!;
+  } else {
+    const metadata = metadataFromPlan(plan);
+    const detailsFromPlan = productDetailsFromPlan(plan, userLocales);
+
+    return {
+      subtitle: detailsFromPlan.subtitle,
+      details: detailsFromPlan.details,
+      successActionButtonLabel: detailsFromPlan.successActionButtonLabel,
+      upgradeCTA: metadata.upgradeCTA,
+    };
+  }
+};
+
+export const productUpgradeFromProductConfig = (
+  plan: AbbrevPlan,
+  useFirestoreProductConfigs: boolean
+) => {
+  if (useFirestoreProductConfigs) {
+    const planConfiguration = getPlanProductConfig(plan);
+    return {
+      productOrder:
+        (planConfiguration.productOrder &&
+          `${planConfiguration.productOrder}`) ||
+        undefined,
+      productSet: planConfiguration.productSet,
+    };
+  } else {
+    const metadata = metadataFromPlan(plan);
+    return {
+      productOrder: metadata.productOrder,
+      productSet: metadata.productSet,
+    };
+  }
+};
```

Comment:
Can you move these new functions to somewhere in shared/subscriptions/configuration/?  I imagine at some point we'd remove the metadata related code in these functions, then we end up with a bunch of product config functions in a file named metadata.ts.

Reply:
Ah yes good idea. I've moved it into `/subscriptions/configuration/helpers.ts` for now, but was thinking `utils.ts` could also be a good place?